packages/flutter/lib/src/material/expansion_panel.dart

```diff
@@ -51,6 +54,15 @@ typedef ExpansionPanelCallback = void Function(int panelIndex, bool isExpanded);
 /// [ExpansionPanel] needs to rebuild.
 typedef ExpansionPanelHeaderBuilder = Widget Function(BuildContext context, bool isExpanded);
 
+/// Signature for the callback that's called when the expansion indicator of the
+/// [ExpansionPanel] needs to rebuild.
+typedef ExpansionPanelIconBuilder = Widget Function(
+  BuildContext context,
+  bool isExpanded,
+  VoidCallback handlePressed,
```

Comment:
it looks like we provide an additional way to trigger the expand. I wonder if we can totally ignore the pointer event of the ExpansionPanelIconBuilder and we just wrap it with our own gesture detector so that we can avoid using this? This will work if user want to use a widget that is not clickable

Reply:
I was speaking with Hans about this last week offline, but we were discussing the following use cases:
1) Two static icons (switching between them depending on whether the widget is expanded or not)
2) AnimatedIcons (which is where `animationDuration` may come in handy)
3) Widgets like Checkbox with their own animations and gesture detectors
4) A single static icon (will need its own gesture detector)

For the first case (assuming that there isn't a GestureDetector), something like the following is needed:
```
expandIconBuilder: (
  BuildContext context,
  bool isExpanded,
  VoidCallback handlePressed,
  Duration animationDuration,
) {
  if (isExpanded) {
    return Padding(
      padding: const EdgeInsets.all(12.0),
      child: InkWell(
        onTap: () { handlePressed(); },
        child: Icon(Icons.check_box),
      ),
    );
  } else {
    return Padding(
      padding: const EdgeInsets.all(12.0),
      child: InkWell(
        onTap: () { handlePressed(); },
        child: Icon(Icons.check_box_outline_blank),
      ),
    );
  }
},
```
We optimize for cases 1 and 4 if we provide our own `GestureDetector`, but we still need a way to allow for the gestures for widgets like `Checkbox` to continue working. Also, if the developer would like to use something like an `InkWell`, providing the `GestureDetector` might not make the most sense.

Reply:
I think it's difficult because there are a variety of possibilities for how the expansion icon can be built/designed, so maybe it's best to leave it to the developer to do the right thing, and the best we can do is be very explicit about how to properly use the builder.

Reply:
Done